span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java

```diff
@@ -32,9 +34,10 @@ public void init(ProcessorContext context) {
       RawSpan rawSpan = converter.convert(value);
       if (null != rawSpan) {
         String traceId = HexUtils.getHex(rawSpan.getTraceId());
-        // we use the trace_id as the key so that raw_span_grouper
+        String customerId = rawSpan.getCustomerId();
+        // we use the (customer_id|trace_id) as the key so that raw_span_grouper
         // job can do a groupByKey without having to create a repartition topic
-        return new KeyValue<>(traceId, rawSpan);
+        return new KeyValue<>(generateSpanKey(customerId, traceId), rawSpan);
```

Comment:
It would be good to define a separate SpanIdentity Avro object containing (customerId, traceId) and use that as the key.

Reply:
+1

Reply:
the grouper should be able to handle this change, right? i.e it's backwards compatible?

Reply:
FYI, sessionization already uses similar Avro class (customerId, sessionId) as the key: https://github.com/Traceableai/sessionization/blob/master/streaming-sessionizer-api/src/main/avro/sessions/SessionIdentity.avdl

I have added similar class in trace-enricher as well.

Reply:
Using TraceIdentity

Reply:
changes have been made to grouper as well